python_scripts/one_time_scripts/update_agentids.py

```diff
@@ -0,0 +1,203 @@
+        if index == 0:
+            record["primary_identifier"] = True
+        else:
+            record["primary_identifier"] = False
+        index += 1
+    return object_json
+
+def main(excel_path, object_type, dry_run=False):
+    """
+    Takes an Excel file of agent IDs, adds the IDs to the agent JSON as Record IDs, then posts it via the ASpace API
+
+    Args:
+        excel_path (str): filepath of the Excel file with the agent IDs
+        object_type (str): the type of object to be updated. In this case, "agents"
+        dry_run (bool): if True, it prints the changed object_json but does not post the changes to ASpace or in the
+        jsonlines file
+    """
+    original_agent_json_data = Path('../../logs', f'update_agentids_original_data_{time.strftime("%Y-%m-%d")}.jsonl')
+    local_aspace = ASpaceAPI(os.getenv('as_api'), os.getenv('as_un'), os.getenv('as_pw'))
+    # pandas.set_option('display.max_rows', 10000)  # Optional for running the script locally to see all returns
+    agent_excelfile = pandas.ExcelFile(excel_path)
+    combined_and_cleaned_sheet = agent_excelfile.parse("combined and cleaned")
+    combined_and_cleaned_sheet.fillna(0, inplace=True)  # replace missing cell values with 0 to sort out later - source: https://medium.com/swlh/data-science-for-beginners-how-to-handle-missing-values-with-pandas-73db5fcd46ec
+    for row in combined_and_cleaned_sheet.itertuples():
+        uri_parts = row.Aspace_link.split("/")
+        original_agent_json = local_aspace.get_object(object_type, uri_parts[-1])
+        if original_agent_json:
+            updated_object_json = deepcopy(original_agent_json)
+            if row.Wikidata_id and row.Wikidata_id != 0:  # TODO: way to do this recursively? while loop or within function?
+                updated_object_json = add_recordID(row.Wikidata_id, "wikidata", updated_object_json,
+                                                   primary=True)
+            if row.SNAC_id and row.SNAC_id != 0:
+                updated_object_json = add_recordID(str(int(row.SNAC_id)), "snac", updated_object_json)  # converting id to integer to remove extraneous decimal places, then convert to string
+            if row.LCNAF_id and row.LCNAF_id != 0:
+                updated_object_json = add_recordID(row.LCNAF_id, "naf", updated_object_json)
+            if row.ULAN_id and row.ULAN_id != 0:
+                updated_object_json = add_recordID(str(int(row.ULAN_id)), "ulan", updated_object_json)  # converting id to integer to remove extraneous decimal places, then convert to string
+            if row.VIAF_id and row.VIAF_id != 0:
+                if "viaf/" in str(row.VIAF_id):
+                    updated_object_json = add_recordID(str(int(row.VIAF_id[5:])), "viaf", updated_object_json)  # converting id to integer to remove extraneous decimal places, then convert to string
+                # if not row.Wikidata_id and not row.SNAC_id and not row.LCNAF_id and not row.ULAN_id:
```

Comment:
Should this commented out section be removed?

Reply:
It could be removed. I wrote it to print out data about when there was only a VIAF ID, since Mark wanted to know how many agents had just that. I can remove it or add a comment to give it context. Which do you think is better?

Reply:
@crugas If you suspect that the condition that made it useful - that you were trying to get stats on something - is a condition that might pop up again, then sure, leave it in but with a comment explaining what it was used for.  Since this is in `one_time_scripts` I think it should be a safe assumption that you can't just willy-nilly run this again without some careful review and potential updates later down the line, but it is still helpful for our future selves' to know what something was used for and why it is commented out.  But, your call.

Reply:
I'll add a comment explaining its purpose and why it's commented out. Considering we may have to run this script again with the additional agent record IDs the AWHI project wants to add, it might be worth having in case it's useful. Thanks!